### PR TITLE
Migrate to .net8

### PR DESCRIPTION
**Pull request by frederikstonge**
Updated all .net references to the new LTS .net 8.

### CODE REVIEW

**Comment by KillStr3aK** [comment]
the runtime got released and its LTS but received several security patches after the release and its currently on hold for a few weeks just to make sure its ready.

**Comment by JustZerooo** [comment]
How is it now :) .NET 8.0 Good.

File: .github/workflows/cmake-single-platform.yml [hunk 1]
```diff
@@ -168,12 +168,12 @@ jobs:
       - name: Add dotnet runtime
         run: |
           mkdir -p build/linux/addons/counterstrikesharp/dotnet
-          curl -s -L https://download.visualstudio.microsoft.com/download/pr/dc2c0a53-85a8-4fda-a283-fa28adb5fbe2/8ccade5bc400a5bb40cd9240f003b45c/aspnetcore-runtime-7.0.11-linux-x64.tar.gz \
+          curl -s -L https://download.visualstudio.microsoft.com/download/pr/257bdcc7-cbfd-4680-964a-cbe8e9160bca/ac0cbf19d897ba51ae004b4146940a0a/aspnetcore-runtime-8.0.0-linux-x64.tar.gz \
```
Comment (roflmuffin) [inline .github/workflows/cmake-single-platform.yml:171]:
```suggestion
          curl -s -L https://download.visualstudio.microsoft.com/download/pr/8e19b03a-93be-43ae-8cd6-95b89a849572/facbb896d726a2496dd23bcecb28c9e9/aspnetcore-runtime-8.0.1-linux-x64.tar.gz \
```

File: .github/workflows/cmake-single-platform.yml [hunk 2]
```diff
@@ -168,12 +168,12 @@ jobs:
...
           | tar xvz -C build/linux/addons/counterstrikesharp/dotnet
-          mv build/linux/addons/counterstrikesharp/dotnet/shared/Microsoft.NETCore.App/7.0.11/* build/linux/addons/counterstrikesharp/dotnet/shared/Microsoft.NETCore.App/
+          mv build/linux/addons/counterstrikesharp/dotnet/shared/Microsoft.NETCore.App/8.0.0/* build/linux/addons/counterstrikesharp/dotnet/shared/Microsoft.NETCore.App/
           
           mkdir -p build/windows/addons/counterstrikesharp/dotnet
-          curl -s -L https://download.visualstudio.microsoft.com/download/pr/a99861c8-2e00-4587-aaef-60366ca77307/a44ceec2c5d34165ae881600f52edc43/aspnetcore-runtime-7.0.11-win-x64.zip -o dotnet.zip
+          curl -s -L https://download.visualstudio.microsoft.com/download/pr/6df1ec69-9cff-4836-b02a-cf5b09baffaa/9cff973dda0520aa1ed216d67433e37a/aspnetcore-runtime-8.0.0-win-x64.zip -o dotnet.zip
```
Comment (roflmuffin) [inline .github/workflows/cmake-single-platform.yml:176]:
```suggestion
          curl -s -L https://download.visualstudio.microsoft.com/download/pr/4fe2aae0-2d64-47b6-ab89-7af36e696bf7/1327a3ad7c6797951ae0017dfd0efbd5/aspnetcore-runtime-8.0.1-win-x64.zip -o dotnet.zip
```

File: src/scripting/dotnet_host.cpp [hunk 3]
```diff
@@ -100,10 +100,10 @@ bool load_hostfxr()
     namespace css = counterstrikesharp;
 #if _WIN32
     std::wstring buffer =
-        std::wstring(css::widen(base_dir) + L"\\dotnet\\host\\fxr\\7.0.11\\hostfxr.dll");
+        std::wstring(css::widen(base_dir) + L"\\dotnet\\host\\fxr\\8.0.0\\hostfxr.dll");
```
Comment (roflmuffin) [inline src/scripting/dotnet_host.cpp:103]:
```suggestion
        std::wstring(css::widen(base_dir) + L"\\dotnet\\host\\fxr\\8.0.1\\hostfxr.dll");
```

File: src/scripting/dotnet_host.cpp [hunk 4]
```diff
@@ -100,10 +100,10 @@ bool load_hostfxr()
...
     CSSHARP_CORE_INFO("Loading hostfxr from {0}", css::narrow(buffer).c_str());
 #else
-    std::string buffer = std::string(base_dir + "/dotnet/host/fxr/7.0.11/libhostfxr.so");
+    std::string buffer = std::string(base_dir + "/dotnet/host/fxr/8.0.0/libhostfxr.so");
```
Comment (roflmuffin) [inline src/scripting/dotnet_host.cpp:106]:
```suggestion
    std::string buffer = std::string(base_dir + "/dotnet/host/fxr/8.0.1/libhostfxr.so");
```

**Comment by frederikstonge** [comment]
I'm happy to see you work on this :)

**Pull request merged**